pipelines/base/config-dump.yaml

```diff
@@ -0,0 +1,122 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: dump-config
+spec:
+  params:
+  - name: REPO_NAME
+    type: string
+    default: .github
+  - name: BRANCH_NAME
+    type: string
+    default: add-initial-peribolos-config
+  - name: GIT_EMAIL
+    type: string
+  - name: GIT_NAME
+    type: string
+  workspaces:
+  - name: secret-data
+    description: location for installation secret
+  - name: shared-data
+    description: location for passing default branch name to the second step
+  steps:
+    - name: dump-and-push-config
+      image: quay.io/operate-first/opf-toolbox
+      script: |
+        #!/bin/bash
+
+        TOKEN=$( cat $(workspaces.secret-data.path)/token )
+        ORG_NAME=$( cat $(workspaces.secret-data.path)/orgName )
+        REPO_REMOTE="https://$TOKEN@github.com/$ORG_NAME/$(params.REPO_NAME)"
+        DEFAULT_BRANCH=$( curl \
+          -H "Accept: application/vnd.github.v3+json" \
+          -H "Authorization: Bearer $TOKEN" \
+          https://api.github.com/repos/$ORG_NAME/$(params.REPO_NAME) 2>/dev/null \
+          | yq e '.default_branch' - | sed 's/\"//g' )
+        echo $DEFAULT_BRANCH > $(workspaces.shared-data.path)/default_branch
+
+        git config --global user.email "$(params.GIT_EMAIL)"
+        git config --global user.name "$(params.GIT_NAME)"
+
+        mkdir repo
+        cd repo
+
+        # Check if repo has no commits
+        if [ $( git ls-remote --heads $REPO_REMOTE | wc -l ) -eq "0" ]; then
+          echo "Repository empty, creating empty commit"
+          git init -b $DEFAULT_BRANCH
+          git remote add origin $REPO_REMOTE
+          git commit --allow-empty -m "Initial commit"
+          git push --all origin
+        else
+          git clone $REPO_REMOTE .
+        fi
+
+        # Check if branch is already created
+        git show-branch origin/$(params.BRANCH_NAME)
+        if [ $? -eq 0 ]; then
+          echo "Branch already exists"
+          git checkout $(params.BRANCH_NAME)
```

Comment:
The branch here may point to a different point in history, a rebase on top of default branch may be needed. Maybe we should use some dynamic branch name so we avoid conflict instead... So we always check out a new branch. WDYT?

Reply:
Resolved in person.